### ISSUE TITLE
Cleanup job removes not expired tokens

### ISSUE THREAD

**Issue by frantisekrokusek** [doorkeeper-gem/doorkeeper#1612]
Hi folks, seems like the job removes not expired tokens...

### Steps to reproduce

```ruby
token = Doorkeeper::AccessToken.create!(..., expires_in:  nil, created_at: 3.days.ago)

token.expired? = false
token.revoked? = false
```

```bash
bundle exec rake doorkeeper:db:cleanup
# deletes the token
```

### Expected behavior
The rake task should to be deleting the tokens with `expires_in` set to `nil`.

### Actual behavior
The rake task deletes the tokens.

### System configuration
```ruby
# config/initializers/doorkeeper.rb
Doorkeeper.configure do
  ...
  access_token_expires_in 24.hours
  ...
end
```

**Ruby version**: `ruby 3.1.2p20`

#### Solution
We need to 
- exclude the tokens and grants that have expires_in set to nil from the collections deleted in `StaleRecordsCleaner`
- use the expires_in value to compute the query

```ruby
# frozen_string_literal: true

module Doorkeeper
  module Orm
    module ActiveRecord
      # Helper class to clear stale and non-active tokens and grants.
      # Used by Doorkeeper Rake tasks.
      class StaleRecordsCleaner
        # Clears expired records but not the ones with unlimited TTL
        def clean_expired(_ttl)
          @base_scope
            # we do not remove unlimitted TTL tokens
            .where.not(expires_in: nil)
            .where("(now() - created_at) > INTERVAL '1 second' * expires_in")
            .in_batches(&:delete_all)
        end
      end
    end
  end
end
```

**Comment by nbulaj** [comment]
Hi @frantisekrokusek 

Currently as I can see AR StaleRecordsCleaner removes next:


```ruby
def clean_expired(ttl)
  table = @base_scope.arel_table

  @base_scope.where(table[:created_at].lt(Time.current - ttl))
    .in_batches(&:delete_all)
end
```

Yeah, looks like it misses at least the records which has `nil` expires_in (BTW I highly suggest not using such tokens).
Would you mind to create a PR to fix it?

**Comment by nbulaj** [comment]
Fix for tokens with `expires_in: nil` released in a 5.6.4
Any PRs with more improvements are very welcome, thanks!
